File: backend/app/models/video.py
from typing import Optional, Literal
from pydantic import BaseModel, HttpUrl


class VideoInfoRequest(BaseModel):
    url: str


class VideoInfoResponse(BaseModel):
    id: str
    title: str
    duration: int
    thumbnail: str


class QueueRequest(BaseModel):
    url: str
    format: str


class QueueVideo(BaseModel):
    id: str
    url: str
    title: str
    format: str

    status: Literal[
        "pending",
        "processing",
        "completed",
        "failed"
    ]

    progress: int = 0

    size: Optional[str] = "Calculating..."

    filename: Optional[str] = None

File: frontend/src/types/video.ts
export type VideoFormat = "mp3-320" | "mp4-1080" | "mp4-720" | "webm-2k";

export type VideoStatus = "pending" | "processing" | "completed" | "failed";

export interface QueueItem {
  id: string;
  url: string;
  title: string;
  format: VideoFormat;
  status: VideoStatus;
  progress: number; // 0 to 100
  size?: string;
}

export interface CompletedItem {
  id: string;
  title: string;
  format: VideoFormat;
  size: string;
  filename: string;
}
